content/learn/book/ecs/resources/_index.md

```diff
@@ -6,12 +6,296 @@ page_template = "book-section.html"
+        .init_resource::<Score>()
+        // Uses the default() value provided by the manual impl of the Default trait
+        .init_resource::<PlayerSupplies>()
+        // Uses the specific value supplied (Turn::Allied) to insert a resource of type Turn
+        .insert_resource(Turn::Allied)
+        // Sets the value of the standard Bevy resource `WindowDescriptor`,
+        // leaving the unspecified fields as their default value
+        .insert_resource(WindowDescriptor {
+            title: "I am a window!".to_string(),
+            width: 500.,
+            height: 300.,
+            vsync: true,
+            ..default()
+        })
+        .add_plugins(DefaultPlugins)
+        .run()
+}
+```
+
+In rare cases, you may need to add a resource after app startup.
+Generally, this is because other resources or entities must exist to ensure proper initialization.
+For that, we can use the equivalent methods on [`Commands`].
+You can add, overwrite and even [`remove_resource`] dynamically in this way.
+Be careful when removing resources: systems will panic if a resource they are expecting is not found!
+
+Prefer adding resources via the methods on `App` whenever possible.
+It's harder to find the resources that your code needs, and like all commands, commands that insert resources are delayed and only take effect at the end of the current stage.
+
+[`'static`]: https://doc.rust-lang.org/rust-by-example/scope/lifetime/static_lifetime.html
+[`Send + Sync`]: https://doc.rust-lang.org/nomicon/send-and-sync.html
+[`App`]: https://docs.rs/bevy/latest/bevy/app/struct.App.html
+[`World`]: https://docs.rs/bevy/latest/bevy/ecs/world/struct.World.html
+[`insert_resouce`]: https://docs.rs/bevy/latest/bevy/app/struct.AppBuilder.html#method.insert_resource
+[`init_resource`]: https://docs.rs/bevy/latest/bevy/app/struct.AppBuilder.html#method.init_resource
+[`Default`]: https://doc.rust-lang.org/beta/std/default/trait.Default.html
+[`FromWorld`]: https://docs.rs/bevy/latest/bevy/ecs/world/trait.FromWorld.html
+[`NonSend`]: https://docs.rs/bevy/latest/bevy/ecs/system/struct.NonSend.html
+[`Commands`]: https://docs.rs/bevy/latest/bevy/ecs/system/struct.Commands.html
+[`remove_resource`]: https://docs.rs/bevy/latest/bevy/ecs/system/struct.Commands.html#method.remove_resource
+
+## Reading and writing from resources
+
+Once our resources have been added to the app, we can read and write to them from systems using the [`Res`] and [`ResMut`] system parameters.
+Let's take a look at how this works by building a tiny guessing game.
+
+```rust
+use bevy::prelude::*;
+
+fn main() {
+    App::build()
+        .add_plugins(MinimalPlugins)
+        .init_resource::<Secret>()
+        .insert_resource(InputMode::Recording)
+        .add_system(record_secret)
+        // The system ordering here ensures that we don't spy on the input before it's entered
+        .add_system(check_secret.before(check_secret))
```

Comment:
Is this line correct? (I don't know Bevy but to me adding a system before itself seems like it doesn't make sense)

Reply:
Nope, this is a typo :)